src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs

```diff
@@ -731,6 +731,7 @@ protected bool TakeStartLine(SocketInput input)
                     // URI was encoded, unescape and then parse as utf8
                     pathEnd = UrlPathDecoder.Unescape(pathBegin, pathEnd);
                     requestUrlPath = pathBegin.GetUtf8String(pathEnd);
+                    requestUrlPath = PathNormalizer.NormalizeToNFC(requestUrlPath);
```

Comment:
@cesarbs Can always move the `ContainsDotSegments` test out of the `RemoveDotSegments`; post `Unescape` for utf8 and pre-`GetAsciiString` for ascii, use `seek()` and set a flag?

Then do `RemoveDotSegments` based on flag?